pkgs/by-name/zo/zoho-mail/package.nix

```diff
@@ -0,0 +1,22 @@
+{ lib
+, appimageTools
+, fetchurl
+
+}:
```

Comment:
```
inherit (stdenv.hostPlatform) system;
throwSystem = throw "Unsupported system: ${system}";
suffix = {
  x86_64-linux = "linux_x86_64.AppImage";
}.${system} or throwSystem;
```
This can be used to explicitly stop builds for systems other than ` x86_64-linux`

Reply:
I always assumed this section was kept to notify `nix` that the package is only available for the specified platform.

Reply:
The `meta` field of the derivation handles that. Looking at the [manual](https://nixos.org/manual/nixpkgs/stable/#var-meta-platforms), we can see that we can use `platforms` and `badPlatforms` to explicitly stop Hydra from building the package. Doing an explicit throw may mess with Hydra building it, and I haven't tested if this doesn't work for other platforms. I think we should be worrying about Hydra over people building from nixpkgs directly, since that has more potential impact vs. someone knowledgeable enough to be tinkering from nixpkgs directly.